src/resolver.ts

```diff
@@ -48,6 +149,46 @@ export class Resolver {
       ctx: Object.assign({}, this.ctx || {}, opts.ctx || {}),
     });
 
-    return runner.resolve(opts.jsonPointer);
+    return runner.resolve(opts);
   }
 }
+
+const printTree = (nodes: INode[] | INode) => {
```

Comment:
Same with these, suggest moving to separate tree.ts or similar file that we can create tests around.